ext/native/thin3d/VulkanRenderManager.cpp

```diff
@@ -646,6 +646,22 @@ void VulkanRenderManager::CopyFramebuffer(VKRFramebuffer *src, VkRect2D srcRect,
 	_dbg_assert_msg_(G3D, dstPos.x + srcRect.extent.width <= (uint32_t)dst->width, "dstPos + extent x > width");
 	_dbg_assert_msg_(G3D, dstPos.y + srcRect.extent.height <= (uint32_t)dst->height, "dstPos + extent y > height");
 
+	for (int i = (int)steps_.size() - 1; i >= 0; i--) {
+		if (steps_[i]->stepType == VKRStepType::RENDER) {
+			if (steps_[i]->render.framebuffer == src) {
+				if (steps_[i]->render.finalColorLayout == VK_IMAGE_LAYOUT_UNDEFINED) {
+					steps_[i]->render.finalColorLayout = VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL;
```

Comment:
Hm.  Should we set a flag that we found dst, so we don't do two render passes affecting this buffer?

-[Unknown]

Reply:
Ignore me, just saw 1dabccb453746ef2c5f0ef5b2590c33833129215.

-[Unknown]